src/EventSubscriber/RequestSubscriber.php

```diff
@@ -185,14 +169,11 @@ public function onFinishRequest(FinishRequestEvent $event) {
       $this->phpAgent->send();
     }
     catch (Exception $e) {
-      // Notify the user of the error.
-      $this->messenger->addError(
-        $this->t('An error occurred while trying to send the transaction to the Elastic APM server.')
-      );
-
       // Log the error to watchdog.
-      $error = Error::decodeException($e);
-      $this->logger->log($error['severity_level'], '%type: @message in %function (line %line of %file).', $error);
+      $this->logger->error($this->t(
+        'An error occurred while trying to send the transaction to the Elastic APM server. The error was @error.', [
+          '@error' => $e->getMessage(),
+      ]));
```

Comment:
**CHANGE REQUEST**

Errors are never translated as they are for developers only; remove `$this->t` and use `sprintf` to better format the error message. Same above.

```
$this->logger->error(sprintf(
  'An error occurred while trying to send the transaction to the Elastic APM server. The error was: %s.',
  $e->getMessage()
));
```